pymc3/distributions/multivariate.py

```diff
@@ -503,10 +506,9 @@ def __init__(self, n, p, *args, **kwargs):
         self.mean = self.n * self.p
         mode = tt.cast(tt.round(self.mean), 'int32')
         diff = self.n - tt.sum(mode, axis=-1, keepdims=True)
-        inc_bool_arr = tt.as_tensor_variable(lst) < diff
-        mode = tt.inc_subtensor(mode[inc_bool_arr.nonzero()], 1)
-        dec_bool_arr = tt.as_tensor_variable(lst) < -diff
```

Comment:
FYI: removing the decrement will cause this
```
import pymc3 as pm
N = 10
with pm.Model() as model:
    m = pm.Multinomial('m', N, [.26, .26, .26, .22], shape=(4,))
sum = m.distribution.mode.eval().sum()
assert sum == N, N-sum
```
to fail like this
```
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-10-9404ca61e1d5> in <module>()
----> 1 assert sum == N, N-sum

AssertionError: -1
```

Should I add this as a test?

Reply:
Thanks, I think it should be fine now as I added `inc_bool_arr = tt.abs_(diff) > 0`

Reply:
oops, missed that

Reply:
I edited a test for testing the case you mentioned as well.